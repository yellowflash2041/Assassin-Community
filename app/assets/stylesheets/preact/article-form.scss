@import 'variables';

.articleformcontainer {
  padding-top: calc(80px + 0.2vw);
  @media screen and ( min-width: 500px ){
    padding-top: calc(98px + 0.2vw);
  }
  padding-bottom: calc(94px + 2vw);
  background: white;
  background: $light-gray;
}

.articleform {
  margin: 0 auto;
  width: calc(95% - 40px);
  max-width: 880px;
  padding: 0px 20px;
  box-shadow: 5px 5px 0px darken($light-medium-gray, 3%);
  border: 1px solid darken($light-medium-gray, 5%);
  border-radius: 3px;
  background: white;
  position:relative;
  .container{
    border: 1px solid $light-medium-gray;
  }
  .help-guide-title{
    padding-top: 20px;
    h1{
      background: $yellow;
    }
  }
}

.articleform__form {
  display: flex;
  flex-direction: column;
}

.articleform__errors{
  background: $red;
  color: white;
  font-weight: bold;
  margin-top: 12px;
  border-radius: 3px;
  padding: 15px;
}

.articleform__orgsettings{
  padding: 0px;
  padding-left: 0px;
  margin: 20px 2px 0px;
  font-size:17px;
  cursor: pointer;
  user-select: none;
  img{
    height: 28px;
    width: 28px;
    border-radius: 50px;
    vertical-align: -7px;
    margin-right: 4px;
  }
  button {
    background: transparent;
    border-radius: 3px;
    border: 0px;
    font-size: 0.8em;
    font-family: $helvetica-condensed;
    width: 80px;
    border-radius: 100px;
    padding: 1px 0px;
    margin-left: 10px;
    border: 3px solid white;
    &.no{
      background: $light-gray;
      color: $medium-gray;
    }
    &.yes{
      background: darken($green, 18%);
      color: white;
    }
  }
  @media screen and ( min-width: 600px ){
    font-size:23px;
    padding-left: 10px;
    img{
      height: 33px;
      width: 33px;
      vertical-align: -8px;
      margin-right: 8px;
    }
    button {
      width: 120px;
    }
  }
}

.articleform__mainimage{
  padding: 10px 0px;
  text-align: center;
  width: 100%;
  img {
    width: 100%;
    height: calc(42vw - 44px);
    background: #fcfcfc;
    @media screen and ( min-width: 985px ){
      height: 370px;
    }
    display: block;
    cursor: pointer;
    &:hover{
      opacity:0.8;
    }
  }
}

.articleform__title {
  height: 26px;
  width: 99%;
  font-size: 18px;
  word-break: break-word;
  border: 0;
  padding: 20px 0px 15px;
  font-weight: 600;
  @media screen and ( min-width: 600px ){
    font-size: 29px;
    height: 50px;
  }
}
.articleform__detailfields{
  display: flex;
  flex-direction: row;
  position: relative;
}
.articleform__tagswrapper{
  display: inline-block;
  flex-grow: 1;
  float: left;
  padding-right: 15px;
  @media screen and ( min-width: 500px ){
    padding-right: 0px;
  }
}

.articleform__tags {
  height: 40px;
  width: 100%;
  border-radius: 3px;
  padding: 8px 8px 7px;
  @media screen and ( min-width: 500px ){
    width: 98%;
  }
  @media screen and ( min-width: 600px ){
    height: 19px;
    border-radius: 100px;
    padding-left: 17px;
  }
  margin-left: -4px;
  resize: none;
  border: 0;
  font-size: 16px;
  font-family: $monospace;
  font-weight: bold;
  background: $light-gray;
}

.articleform__tagsoptions{
  position: absolute;
  left: 0;
  right: 0;
  top: 50px;
  @media screen and ( min-width: 600px ){
    top: 30px;
  }
  background: white;
  border: 1px solid $medium-gray;
  box-shadow: $bold-shadow;
  font-family: $monospace;
  z-index:20;
}
.articleform__tagoptionrow{
  padding: 10px;
  cursor:pointer;
  &:hover{
    background: lighten($green, 27%);
  }
}
.articleform__tagoptionrow--active{
  background: $green;
  &:hover{
    background: darken($green, 10%);
  }
}

.articleform__detailsButton{
  color: $black;
  background: $purple;
  border: 0px;
  border-radius: 300px;
  padding: 3px 18px;
  font-family: $helvetica-condensed;
  font-size: 14.4px;
  height: 34px;
  display: none;
  @media screen and ( min-width: 600px ){
    display: inline-block;
  }
  margin-left:25px;
  &:hover{
    background: lighten($green, 24%);
  }
  img{
    width: 21px;
    height: 21px;
    vertical-align: -4px;
    display: inline-block;
  }
  &.articleform__detailsButton--moreconfig {
    margin-left: 5px;
    background: $light-medium-gray;
    &.articleform__detailsButton--bottom{
      margin-left: 5px;
      width:50px;
    }
  }
  &.articleform__detailsButton--bottom{
    margin-left: -5px;
    margin-top:10px;
    margin-bottom: 8px;
    display: inline-block;
    padding: 2px 15px;
    border-radius: 3px;
    width:100px;
    @media screen and ( min-width: 600px ){
      display: none;
    }  
  }
}

.articleform__description {
  height: 30px;
}

.articleform__body {
  min-height: calc(90vh - 370px);
  // height: calc(90vh - 240px);
  width: 98%;
  border: 0;
  font-size: 15px;
  resize: none;
  padding: 20px 6px;
  font-family: $monospace;
  @media screen and ( min-width: 600px ){
    font-size: 18px;
    min-height: calc(90vh - 280px);
  }
}

.articleform__buttons{
  padding: 20px 0px 20px;
  text-align: center;
  position:fixed;
  bottom: 0px;
  left: 0;
  right: 0;
  background: white;
  z-index: 20;
  border-top: 1px solid $light-medium-gray;
  button, a{
    cursor:pointer;
    background: $light-gray;
    font-family: $helvetica-condensed;
    font-size:14.5px;
    border: 0px;
    border-radius: 113px;
    box-shadow:  2px 2px 8px darken($light-medium-gray, 13%);
    padding: 6px 13px;
    display: inline-block;
    text-align: center;
    color: $medium-gray;
    margin: 4px calc(3px + 0.7vw);
    &:hover{
      background: lighten($green, 24%);
    }
    &.active{
      background: $green;
    }
    &.articleform__buttons--small{
      width: 45px;
      border: white;
    }
    &.articleform__buttons--publish{
      background: $bold-blue;
      color: white;
      &:hover{
        opacity:0.9;
      }
    }
    @media screen and ( min-width: 600px ){
      font-size:16px;
      width: 130px;
    }
  }
}

.articleform__overlay{
  position: absolute;
  top: 0px;
  left: 0;
  right: 0;
  bottom: 0;
  background: $dark-gray;
  color: white;
  padding: 5%;
  box-shadow: $bold-shadow;
  text-align: center;
  border-radius: 3px;
  h2{
    color: $green;
    font-size: 2em;
    font-family: $helvetica-condensed;
  }
  input[type="file"]{
    width: 90%;
    background: $black;
    padding: 20px;
    border-radius: 3px;
    margin: 20px auto;
    display: block;
    cursor:pointer;
  }
  input[type="text"]{
    width: calc(80% - 20px);
    padding: 10px 20px;
    border-radius: 3px;
    margin: 5px auto 0px;
    display: inline-block;
    border: 0px;
    font-size: 1em;
    @media screen and ( min-width: 600px ){
      width: calc(100% - 270px);
      border-top-left-radius: 0px;
      border-bottom-left-radius: 0px;
      margin: 20px auto 0px;
    }
  }
  label {
    margin-top:20px;
    display: inline-block;
    width: 80%;
    font-weight: bold;
    color: white;
    font-size: 1em;
    background: $medium-gray;
    padding: 10px;
    border-radius: 3px;
    @media screen and ( min-width: 600px ){
      width: 180px;
      border-top-right-radius: 0px;
      border-bottom-right-radius: 0px;  
    }

  }
  small {
    display: block;
    font-size:0.8em;
    color: $light-medium-gray;
    text-align: left;
    width: 80%;
    margin: 7px auto;
    font-style:oblique;
    @media screen and ( min-width: 600px ){
      width: calc(100% - 40px);
    }
    button {
      margin-left: 5px;
      background: $bold-blue;
    }
  }
  .articleform__exitbutton{
    position:absolute;
    color: white;
    font-size: 37px;
    font-weight: bold;
    left: 1%;
    top: 0%;
    background: transparent;
    border: 0px;
    cursor:pointer;
  }
  button{
    background: $red;
    color: white;
    border: 0px;
    padding: 4px 10px;
    font-size: 14px;
    border-radius: 3px;
    font-weight: bold;
  }
  .articleform__donebutton{
    display: block;
    font-size: 22px;
    background: $green;
    color: $black;
    padding: 10px 40px;
    margin: 20px auto;

  }
  img{
    display: block;
    margin: 10px auto;
    max-width: 80%;
  }
}

.articleform__notice{
  position: fixed;
  bottom:0;
  left:0;
  right:0;
  padding: 25px 0px;
  color: white;
  background: $green;
  text-align: center;
  font-size:2em;
  font-weight: bold;
  z-index:50;
  &.articleform__notice--draft{
    background: $yellow;
    color: $black;
  }
}