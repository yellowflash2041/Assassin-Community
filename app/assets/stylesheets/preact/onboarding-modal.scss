@import 'variables';
@import 'mixins';

.pages-onboarding {
  min-height: initial;
}

.onboarding-body {
  @include themeable(background-color, theme-background, $white);
  position: fixed;
  margin-top: 50px;
  bottom: 0;
  height: 100vh;
  width: 100%;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
  line-height: 200%;
  z-index: 11;
  margin: 0;
  background-image: url('/assets/onboarding-background-white.png');
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  /* Center the image */
  background-position: center;
  /* Do not repeat the image */
  background-repeat: no-repeat;

  a {
    @include themeable(color, theme-anchor-color, #2e3b9e);
  }

  .sticker-logo {
    height: 80px;
    width: 80px;
    transform: rotate(-5deg);
    padding: 10px;
    vertical-align: middle;
    top: -10px;
    padding-top: 5px;
  }

  .onboarding-main {
    display: flex;
    flex-direction: column;
    width: 800px;
    height: 466px;
    @include themeable(background-color, theme-container-background, $white);
    border: 8px solid #6b55c9;
    @include themeable(border-color, theme-container-border, #6b55c9);
    padding: 30px;
    border-radius: 3px;
    z-index: 1;
    font-size: 20px;
    box-shadow: 0 0 8px 2px rgba(0, 0, 0, 0.3), 8px 0px 13px rgba(0, 0, 0, 0.21);
    align-items: center;
    justify-content: center;
  }

  .onboarding-content {
    flex: 1 0 auto;
    width: 100%;
    height: 440px;
    margin-top: 5px;

    h1 {
      margin-top: 15px;
      line-height: 1.1em;
    }
  }

  .onboarding-navigation {
    align-self: flex-end;
    flex-shrink: 0;
    width: 100%;
  }

  .background-image {
    z-index: -1;
    position: absolute;
    top: 60px;
    left: 0px;
  }

  h1 {
    text-align: center;
  }

  h2 {
    font-size: 15px;

    @media screen and (min-width: 600px) {
      font-size: 26px;
    }
  }

  p {
    line-height: 1.25em;
  }

  button {
    width: 130px;
    padding: 6px 0px;
    height: auto;
    margin: 11px 20px;
    text-align: center;
    font-weight: bold;
    border-radius: 3px;
    border: 2px solid #0a0a0a;
    @include themeable(border-color, theme-color, #0a0a0a);
    @include themeable(color, theme-color, #0a0a0a);
    @include themeable(background-color, theme-container-accent-background, $light-gray);
    font-family: 'HelveticaNeue-CondensedBold', 'HelveticaNeueBoldCondensed',
      'HelveticaNeue-Bold-Condensed', 'Helvetica Neue Bold Condensed',
      'HelveticaNeueBold', 'HelveticaNeue-Bold', 'Helvetica Neue Bold',
      'HelveticaNeue', 'Helvetica Neue', 'TeXGyreHerosCnBold', 'Helvetica',
      'Tahoma', 'Geneva', 'Arial Narrow', 'Arial', sans-serif;
    -webkit-appearance: none;
    font-stretch: condensed;
    font-size: 26px;
  }

  .next-button {
    @include themeable(background, theme-top-bar-write-background, #66e2d5);
    float: right;
  }

  .about {
    width: 100%;

    label,
    input,
    textarea {
      display: block;
      width: 95%;
      margin: 0 auto;
    }

    input,
    textarea {
      padding: 12px;
      font-size: 19px;
      text-emphasis: bold;
      border: 1px solid #dee6e9;
      border-radius: 3px;
    }

    textarea {
      height: 120px;
    }
  }

  .checkbox-slide {
    label {
      display: block;
      cursor: pointer;
    }

    input[type='checkbox'] {
      -webkit-appearance: none;
      width: 19px;
      height: 19px;
      background: transparent;
      border-radius: 5px;
      border: 2px solid #555;
      vertical-align: middle;
      margin: 8px;
      cursor: pointer;
    }

    input[type='checkbox']:checked {
      @include themeable(background, theme-anchor-color, #66e2d5);
    }
  }

  .warning-message {
    background-color: red;
    opacity: 0.7;
    padding: 10px;
    width: 100%;
    border-radius: 3px;
    color: white;
  }

  .scroll {
    height: 365px;
    overflow: auto;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    border-bottom: 1px solid $light-medium-gray;
  }

  .tag {
    padding: 44px 6px 46px;
    font-size: 20px;
    margin: 5px;
    border-radius: 3px;
    font-weight: bold;
    border: none;
    outline: none;
    overflow: hidden;
    position: relative;
    text-overflow: ellipsis;
    width: 100%;
    opacity: 0.8;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
      Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji',
      'Segoe UI Symbol';

    @media screen and (min-width: 600px) {
      width: 255px;
    }

    &.tag-selected {
      opacity: 1;
    }

    &:hover {
      opacity: 1;
    }

    .onboarding-tag-follow-indicator {
      position: absolute;
      bottom: 22px;
      left: 0px;
      right: 0px;
      text-align: center;
      font-size: 0.66em;
    }
  }

  .select-all-button-wrapper {
    width: 100%;

    button {
      width: 300px;
      line-height: 1em;
      text-align: left;
      padding-left: 15px;
    }
  }

  .user {
    width: calc(100% - 40px);
    overflow: hidden;
    text-overflow: ellipsis;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
      Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji',
      'Segoe UI Symbol';
    font-size: 15px;
    padding: 5px;
    border-radius: 3px;
    text-align: center;
    margin-right: 5px;
    position: relative;
    color: $black;

    .onboarding-user-follow-status {
      position: absolute;
      color: #32a99c;
      right: 8px;
      top: 5px;
      font-size: 0.88em;
    }

    span {
      display: block;
      width: 100%;
    }

    p {
      font-weight: 400;
      height: 55px;
    }

    @media screen and (min-width: 600px) {
      width: 365px;
    }

    img {
      height: 80px;
      width: 80px;
      border-radius: 50%;
      padding: 5px;
      margin: 10px auto;
      display: block;
    }
  }

  .user:active {
    background-color: red;
  }

  .badge-image {
    position: absolute;
  }

  .onboarding-what-next {
    display: flex;
    flex-direction: column;
    height: 250px;
    overflow: auto;

    @media screen and (min-width: 600px) {
      flex-direction: row;
    }

    a {
      background: #d9e7ff;
      padding: 15px 4px 0px 12px;
      height: 140px;
      margin: 4px auto;
      border-radius: 3px;
      width: 350px;
      display: block;
      line-height: 1.2em;
      font-weight: bold;
      position: relative;
      text-align: center;

      @media screen and (min-width: 600px) {
        height: 220px;
        margin: 4px 4px;
        text-align: left;
        padding: 20px 4px 0px 12px;
      }

      .whatnext-emoji {
        display: none;
        text-align: center;
        position: absolute;
        bottom: 22px;
        left: 0;
        right: 0;
        font-size: 50px;

        @media screen and (min-width: 600px) {
          display: block;
        }
      }
    }
  }

  .onboarding-previous-location {
    text-align: center;
    margin-top: 10px;
    font-weight: bold;
    display: none;

    @media screen and (min-width: 600px) {
      display: block;
    }

    code {
      font-size: 0.6em;
      margin-top: -10px;
      display: block;
      font-weight: 400;
    }
  }

  @media screen and (max-width: 800px) {
    .onboarding-main {
      width: 100%;
      height: 100%;
      background-image: url('/assets/mobile-onboarding-background.png');
      background-size: cover;
      /* Center the image */
      background-position: center;
      /* Do not repeat the image */
      background-repeat: no-repeat;
      border: none;
      color: $black;
    }

    .onboarding-content {
      flex: 0 0 auto;

      h1,
      h2 {
        margin-top: initial;
      }
    }

    span {
      display: block;
    }
  }

  @media screen and (max-width: 650px) {
    .onboarding-main {
      font-size: 18px;
    }

    button {
      width: 100px;
      font-size: 20px;
    }
  }

  @media screen and (max-width: 450px) {
    .onboarding-main {
      font-size: 16px;
    }

    button {
      width: 100px;
      font-size: 18px;
    }
  }

  @media screen and (max-width: 330px) {
    button {
      width: 80px;
      font-size: 14px;
    }

    .sticker-logo {
      height: 40px;
      width: 40px;
    }

    h1 {
      font-size: 20px;
    }
  }
}