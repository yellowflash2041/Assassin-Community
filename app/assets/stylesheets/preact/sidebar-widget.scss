@import 'variables';

.sidebar-additional{
  .widget{
    .widget-suggested-follows-container{
      h4{
        margin: 0;
      }
      .widget-body{
        .widget-list-item__suggestions{
          .widget-list-item__close-button{
            float: right;
            width: 24px;
            height: 19px;
            margin: 3px;
          }
          .widget-list-item__content{
            max-width: 180px;
            margin-left: 60px;
            margin-right: 20px;
            min-height: 55px;
            a{
              display: block;
              padding: 5px 0px;
              font-size: 18px;
              color: $black;
              color: var(--theme-color, $black);
              &:hover {
                text-decoration: underline;
              }
            }
            img{
              width: 48px;
              height: 48px;
              border-radius: 50%;
              position: absolute;
              top: unset;
              left: 15px;
            }
            .widget-list-item__follow-button{
              border: 2px solid transparent;
              border-radius: 3px;
              padding: 2px 8px;
              text-align: center;
              background-color: $green;
              color: $black;
              font-family: $helvetica-condensed;
              font-weight: 900;
              font-size: 0.8em;
            }
          }
        }
      }
    }
  }
}
