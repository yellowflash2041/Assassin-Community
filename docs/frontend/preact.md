---
title: Preact
---

# Preact

[Preact](https://preactjs.com/) is an alternative to React with the same modern API.

Preact components are packaged using [Webpacker](/frontend/webpacker) and the Preact code is located in `app/javascript`.

The components are mounted when needed, look for `javascript_pack_tag` in the view pages inside `app/views`.
