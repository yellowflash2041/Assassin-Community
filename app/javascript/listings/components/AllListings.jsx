import { h } from 'preact';
import PropTypes from 'prop-types';
import SingleListing from '../singleListing';

const AllListings = ({
  listings,
  onAddTag,
  onChangeCategory,
  currentUserId,
  onOpenModal,
}) => {
  return (
    <div className="classifieds-columns" id="classified-listings-results">
      {listings.map((listing) => (
        <SingleListing
          onAddTag={onAddTag}
          onChangeCategory={onChangeCategory}
          listing={listing}
          currentUserId={currentUserId}
          onOpenModal={onOpenModal}
          isOpen={false}
        />
      ))}
    </div>
  );
};

AllListings.propTypes = {
  currentUserId: PropTypes.number,
  listings: PropTypes.isRequired,
  onAddTag: PropTypes.func.isRequired,
  onChangeCategory: PropTypes.func.isRequired,
  onOpenModal: PropTypes.func.isRequired,
};

AllListings.defaultProps = {
  currentUserId: null,
};

export default AllListings;
