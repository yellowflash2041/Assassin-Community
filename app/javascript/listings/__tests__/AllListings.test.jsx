import { h } from 'preact';
import { render } from '@testing-library/preact';
import { axe } from 'jest-axe';
import AllListings from '../components/AllListings';

const firstListing = {
  id: 20,
  category: 'misc',
  location: 'West Refugio',
  processed_html:
    '\u003cp\u003eEius et ullam. Dolores et qui. Quis \u003cstrong\u003equi\u003c/strong\u003e omnis.\u003c/p\u003e\n',
  slug: 'illo-iure-quos-htyashsayas-5hk7',
  title: 'Mentor wanted',
  tags: ['go', 'git'],
  user_id: 1,
  author: {
    name: 'Mrs. Yoko Christiansen',
    username: 'mrschristiansenyoko',
    profile_image_90:
      '/uploads/user/profile_image/7/4b1c980a-beb0-4a5f-b3f2-acc91adc503c.png',
  },
};

const secondListing = {
  id: 21,
  category: 'misc',
  location: 'West Refugio',
  processed_html:
    '\u003cp\u003eEius et ullam. Dolores et qui. Quis \u003cstrong\u003equi\u003c/strong\u003e omnis.\u003c/p\u003e\n',
  slug: 'illo-iure-quos-ereerr-5hk7',
  title: 'This is an awesome listing',
  tags: ['functional', 'clojure'],
  user_id: 1,
  author: {
    name: 'Mr. Rogers',
    username: 'fred',
    profile_image_90:
      '/uploads/user/profile_image/7/4b1c980a-beb0-4a5f-b3f2-acc91adc503c.png',
  },
};

const thirdListing = {
  id: 22,
  category: 'misc',
  location: 'West Refugio',
  processed_html:
    '\u003cp\u003eBobby says hello. Eius et ullam. Dolores et qui. Quis \u003cstrong\u003equi\u003c/strong\u003e omnis.\u003c/p\u003e\n',
  slug: 'illo-iure-fss-ssasas-5hk7',
  title: 'Illo iure quos perspiciatis',
  tags: ['twitter', 'learning'],
  user_id: 1,
  author: {
    name: 'Mrs. John Mack',
    username: 'mrsjohnmack',
    profile_image_90:
      '/uploads/user/profile_image/7/4b1c980a-beb0-4a5f-b3f2-acc91adc503c.png',
  },
};

const listings = [firstListing, secondListing, thirdListing];

const getProps = () => ({
  listings,
  onAddTag: () => {
    return 'onAddTag';
  },
  onChangeCategory: () => {
    return 'onChangeCategory';
  },
  currentUserId: 1,
  message: 'Something',
  onOpenModal: () => {
    return 'onSubmit;';
  },
});

const renderAllListings = () => render(<AllListings {...getProps()} />);

describe('<AllListings />', () => {
  it('should have no a11y violations', async () => {
    const { container } = renderAllListings();
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });

  it('should render the given listings', async () => {
    const { getByTestId, getByText } = renderAllListings();

    // Ensure each listing is present
    const titleOptions = {
      selector: 'h3 > a',
    };

    // 1st listings
    getByTestId('single-listing-20');

    // listing title
    const listing1Title = getByText('Mentor wanted', titleOptions);

    expect(listing1Title.getAttribute('href')).toEqual(
      '/listings/misc/illo-iure-quos-htyashsayas-5hk7',
    );

    // listing body
    getByText(/Eius et ullam. Dolores et qui. Quis/, {
      selector: '[data-testid="single-listing-20"] *',
    });

    // listing tags
    const goTag = getByText('go', {
      selector: '[data-testid="single-listing-20"] a',
    });

    expect(goTag.getAttribute('href')).toEqual('/listings?t=go');

    const gitTag = getByText('git', {
      selector: '[data-testid="single-listing-20"] a',
    });

    expect(gitTag.getAttribute('href')).toEqual('/listings?t=git');

    // listing author
    const listing1Author = getByText('Mrs. Yoko Christiansen', {
      selector: '[data-testid="single-listing-20"] a',
    });

    expect(listing1Author.getAttribute('href')).toEqual('/mrschristiansenyoko');

    // 2nd listing
    getByTestId('single-listing-21');

    // listing title
    const listing2Title = getByText('This is an awesome listing', titleOptions);

    expect(listing2Title.getAttribute('href')).toEqual(
      '/listings/misc/illo-iure-quos-ereerr-5hk7',
    );

    // listing body
    getByText(/Eius et ullam. Dolores et qui. Quis/, {
      selector: '[data-testid="single-listing-21"] *',
    });

    // listing tags
    const functionalTag = getByText('functional', {
      selector: '[data-testid="single-listing-21"] a',
    });

    expect(functionalTag.getAttribute('href')).toEqual(
      '/listings?t=functional',
    );

    const clojureTag = getByText('clojure', {
      selector: '[data-testid="single-listing-21"] a',
    });

    expect(clojureTag.getAttribute('href')).toEqual('/listings?t=clojure');

    // listing author
    const listing2Author = getByText('Mr. Rogers', {
      selector: '[data-testid="single-listing-21"] a',
    });

    expect(listing2Author.getAttribute('href')).toEqual('/fred');

    // 3rd listing
    getByTestId('single-listing-22');

    // listing title
    const listing3Title = getByText(
      'Illo iure quos perspiciatis',
      titleOptions,
    );

    expect(listing3Title.getAttribute('href')).toEqual(
      '/listings/misc/illo-iure-fss-ssasas-5hk7',
    );

    // listing body
    getByText(/Bobby says hello. Eius et ullam. Dolores et qui. Quis/, {
      selector: '[data-testid="single-listing-22"] *',
    });

    // listing tags
    const twitterTag = getByText('twitter', {
      selector: '[data-testid="single-listing-22"] a',
    });

    expect(twitterTag.getAttribute('href')).toEqual('/listings?t=twitter');

    const learningTag = getByText('learning', {
      selector: '[data-testid="single-listing-22"] a',
    });

    expect(learningTag.getAttribute('href')).toEqual('/listings?t=learning');

    const listing3Author = getByText('Mrs. John Mack', {
      selector: '[data-testid="single-listing-22"] a',
    });

    expect(listing3Author.getAttribute('href')).toEqual('/mrsjohnmack');
  });
});
