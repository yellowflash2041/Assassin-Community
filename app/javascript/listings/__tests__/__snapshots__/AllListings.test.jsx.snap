// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllListings /> Should render all listings 1`] = `
<div
  class="classifieds-columns"
  id="classified-listings-results"
>
  <div
    class="single-classified-listing"
    id="single-classified-listing-20"
  >
    <div
      class="listing-content"
    >
      <h3
        class="single-classified-listing-header"
      >
        <a
          data-listing-id={20}
          data-no-instant={true}
          href="/listings/misc/illo-iure-quos-htyashsayas-5hk7"
          onClick={[Function]}
        >
          Mentor wanted
        </a>
        <div
          class="dropdown-menu"
          ref={Object {}}
        >
          <button
            aria-label="Toggle dropdown menu"
            class="dropdown-btn"
            onClick={[Function]}
            type="button"
          >
            <img
              alt="Dropdown menu icon"
              class="dropdown-icon"
              src=""
            />
          </button>
          <div
            class="dropdown"
          >
            <div
              class="dropdown-content "
            >
              <a
                class="classified-listing-edit-button"
                href="/listings/20/edit"
              >
                Edit
              </a>
            </div>
          </div>
        </div>
      </h3>
      <div
        class="single-classified-listing-body"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Eius et ullam. Dolores et qui. Quis <strong>qui</strong> omnis.</p>
",
          }
        }
      />
      <div
        class="single-classified-listing-tags"
      >
        <a
          data-no-instant={true}
          href="/listings?t=go"
          onClick={[Function]}
        >
          go
        </a>
        <a
          data-no-instant={true}
          href="/listings?t=git"
          onClick={[Function]}
        >
          git
        </a>
      </div>
      <div
        class="single-classified-listing-author-info"
      >
        <a
          data-no-instant={true}
          href="/listings/misc"
          onClick={[Function]}
        >
          misc
        </a>
        <a
          href="/listings/?q=West Refugio"
        >
          ・West Refugio
        </a>
        ・
        <a
          href="/mrschristiansenyoko"
        >
          Mrs. Yoko Christiansen
        </a>
      </div>
    </div>
  </div>
  <div
    class="single-classified-listing"
    id="single-classified-listing-21"
  >
    <div
      class="listing-content"
    >
      <h3
        class="single-classified-listing-header"
      >
        <a
          data-listing-id={21}
          data-no-instant={true}
          href="/listings/misc/illo-iure-quos-ereerr-5hk7"
          onClick={[Function]}
        >
          Second tag.
        </a>
        <div
          class="dropdown-menu"
          ref={Object {}}
        >
          <button
            aria-label="Toggle dropdown menu"
            class="dropdown-btn"
            onClick={[Function]}
            type="button"
          >
            <img
              alt="Dropdown menu icon"
              class="dropdown-icon"
              src=""
            />
          </button>
          <div
            class="dropdown"
          >
            <div
              class="dropdown-content "
            >
              <a
                class="classified-listing-edit-button"
                href="/listings/21/edit"
              >
                Edit
              </a>
            </div>
          </div>
        </div>
      </h3>
      <div
        class="single-classified-listing-body"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Eius et ullam. Dolores et qui. Quis <strong>qui</strong> omnis.</p>
",
          }
        }
      />
      <div
        class="single-classified-listing-tags"
      >
        <a
          data-no-instant={true}
          href="/listings?t=functional"
          onClick={[Function]}
        >
          functional
        </a>
        <a
          data-no-instant={true}
          href="/listings?t=clojure"
          onClick={[Function]}
        >
          clojure
        </a>
      </div>
      <div
        class="single-classified-listing-author-info"
      >
        <a
          data-no-instant={true}
          href="/listings/misc"
          onClick={[Function]}
        >
          misc
        </a>
        <a
          href="/listings/?q=West Refugio"
        >
          ・West Refugio
        </a>
        ・
        <a
          href="/mrschristiansenyoko"
        >
          Mrs. Ashahir
        </a>
      </div>
    </div>
  </div>
  <div
    class="single-classified-listing"
    id="single-classified-listing-22"
  >
    <div
      class="listing-content"
    >
      <h3
        class="single-classified-listing-header"
      >
        <a
          data-listing-id={22}
          data-no-instant={true}
          href="/listings/misc/illo-iure-fss-ssasas-5hk7"
          onClick={[Function]}
        >
          Illo iure quos perspiciatis.
        </a>
        <div
          class="dropdown-menu"
          ref={Object {}}
        >
          <button
            aria-label="Toggle dropdown menu"
            class="dropdown-btn"
            onClick={[Function]}
            type="button"
          >
            <img
              alt="Dropdown menu icon"
              class="dropdown-icon"
              src=""
            />
          </button>
          <div
            class="dropdown"
          >
            <div
              class="dropdown-content "
            >
              <a
                class="classified-listing-edit-button"
                href="/listings/22/edit"
              >
                Edit
              </a>
            </div>
          </div>
        </div>
      </h3>
      <div
        class="single-classified-listing-body"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Eius et ullam. Dolores et qui. Quis <strong>qui</strong> omnis.</p>
",
          }
        }
      />
      <div
        class="single-classified-listing-tags"
      >
        <a
          data-no-instant={true}
          href="/listings?t=twitter"
          onClick={[Function]}
        >
          twitter
        </a>
        <a
          data-no-instant={true}
          href="/listings?t=learning"
          onClick={[Function]}
        >
          learning
        </a>
      </div>
      <div
        class="single-classified-listing-author-info"
      >
        <a
          data-no-instant={true}
          href="/listings/misc"
          onClick={[Function]}
        >
          misc
        </a>
        <a
          href="/listings/?q=West Refugio"
        >
          ・West Refugio
        </a>
        ・
        <a
          href="/mrschristiansenyoko"
        >
          Mrs. John Mack
        </a>
      </div>
    </div>
  </div>
</div>
`;
